nats.go

```diff
@@ -3306,19 +3306,20 @@ func NewMsg(subject string) *Msg {
 }
 
 const (
-	hdrLine            = "NATS/1.0\r\n"
-	crlf               = "\r\n"
-	hdrPreEnd          = len(hdrLine) - len(crlf)
-	statusHdr          = "Status"
-	descrHdr           = "Description"
-	lastConsumerSeqHdr = "Nats-Last-Consumer"
-	lastStreamSeqHdr   = "Nats-Last-Stream"
-	consumerStalledHdr = "Nats-Consumer-Stalled"
-	noResponders       = "503"
-	noMessagesSts      = "404"
-	reqTimeoutSts      = "408"
-	controlMsg         = "100"
-	statusLen          = 3 // e.g. 20x, 40x, 50x
+	hdrLine              = "NATS/1.0\r\n"
+	crlf                 = "\r\n"
+	hdrPreEnd            = len(hdrLine) - len(crlf)
+	statusHdr            = "Status"
+	descrHdr             = "Description"
+	lastConsumerSeqHdr   = "Nats-Last-Consumer"
+	lastStreamSeqHdr     = "Nats-Last-Stream"
+	consumerStalledHdr   = "Nats-Consumer-Stalled"
+	noResponders         = "503"
+	noMessagesSts        = "404"
+	reqTimeoutSts        = "408"
+	leadershipChangedSts = "409"
```

Comment:
I would refrain doing that, since server tends to use the same code for different errors (case in point 408 has several descriptions at the moment). So status 409 is not necessarily leadership change. For instance trying to do a "next message" to a push consumer would return "Consumer is push based", or doing a a Fetch() with a batch higher that MaxRequestBatch would return "Exceeded MaxRequestBatch of <x>", etc...